app/controllers/all-events.js

```diff
@@ -0,0 +1,320 @@
+            }
+            this.set("model.eventShown", sorted);
+        },
+
+        toggleRequestSent(item) {
+
+
+
+            let lastForm = this.get("lastForm");
+
+            if (lastForm && $(item.target).hasClass("onclickOrange")) {
+                let elements = $(lastForm).nextAll();
+                for (var i = 0; i < 3; i++) {
+                    if ($(window).innerWidth() < 768) {
+                        $(elements[i]).slideUp();
+                        console.log("twice");
+                    }
+                }
+                let form = $(item.target);
+                this.set("lastForm", form);
+            } else {
+                if (lastForm) {
+                    $(lastForm).removeClass("onclickOrange");
+
+                    let elements = $(lastForm).nextAll();
+                    for (var i = 0; i < 3; i++) {
+
+                        if ($(window).innerWidth() < 768) {
+                            $(elements[i]).slideUp();
+                            console.log("once");
+                        }
+                    }
+                }
+                let form = $(item.target);
+                $(form).addClass("onclickOrange");
+                let elements = $(form).nextAll();
+                for (var i = 0; i < 3; i++) {
+                    if ($(window).innerWidth() < 768) {
+                        $(elements[i]).slideDown();
+                        console.log("twice");
+                    }
+                }
+                this.set("lastForm", form);
+            }
+        },
+
+        filterEvents: function() {
+            // Filter the list of activities shown when user types in the search bar
+            let searchValue = this.get("model.searchValue");
+            if (searchValue) {
+                let newList = [];
+                let oldList = this.get("model.allEvents");
+                console.log(oldList);
+                for (let i = 0; i < oldList.length; i++) {
+                    if (oldList[i].Event_Name.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1) {
+                        newList.push(oldList[i]);
+                    } else if (oldList[i].End_Time.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1) {
+                        newList.push(oldList[i]);
+                    } else if (oldList[i].Location !== null) {
+                        if (oldList[i].Location.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1) {
+                            newList.push(oldList[i]);
+                        }
+                    } else if (oldList[i].Start_Time.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1) {
+                        newList.push(oldList[i]);
+                    }
+
+                }
+                this.set("model.eventShown", newList);
+            } else {
+                this.set("model.eventShown", this.get("model.allEvents"));
+            }
+        },
+
+
+        toggleCheckBox: function() {
```

Comment:
This would probably be easier if you put the checkboxes in a form and had the form auto-submit on change.  Then you might have easier access to the values.